### PR TITLE
HSD8-129 Call video plugins to get video title for entity name

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Populate embed code titles based on bundle suggestion plugins. for videos, it then calls the video provider plugins to get the titles. see https://www.drupal.org/project/video_embed_field/issues/2913925
- Fixed max file size issue when media bundle is configured to prevent certain file sizes.

# Needed By (Date)
- Some Day

# Urgency
- low

# Steps to Test

1. checkout latest (suhumsci repo)[https://github.com/SU-HSDO/suhumsci]
1. composer update/install `composer install --prefer-source`
1. sync a site `blt drupal:sync --site=[sitename]`
1. in any wysiwyg click the media browser button.
1. on the embed tab, enter a youtube url
1. after saving, ensure the entity title is the same as the youtube title.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
poke

**Comment by sherakama** [comment]
Ah, sorry, this one fell off my radar. I thought I got to them all last Friday. I'll review a little later today.

**Comment by sherakama** [comment]
Lol, I'm not sure but it feels like it should be against the rules to poke and then quickly push a couple of new commits.

**Comment by pookmish** [comment]
Yeah definitely a bad thing. but found some logic that is currently breaking the live site. Users can currently attempt to upload up to 256MB files. not a good thing

File: src/Plugin/BundleSuggestionManager.php [hunk 1]
```diff
@@ -71,24 +85,170 @@ public function getDefinitions() {
...
+    foreach ($media_types as $media_type) {
+      $extensions = array_merge($extensions, $this->getBundleExtensions($media_type));
+    }
+    return array_unique(array_filter($extensions));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMaxFilesize(array $bundles = []) {
```
Comment (sherakama) [inline src/Plugin/BundleSuggestionManager.php]: This should be a capital "S" on size.

Comment (pookmish) [reply src/Plugin/BundleSuggestionManager.php]: doh. thought i had that

File: src/Plugin/BundleSuggestion/VideoEmbed.php [hunk 2]
```diff
@@ -60,10 +60,18 @@ public function getBundleFromString($input) {
...
       }
     }
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getName($input) {
+    $video_provider = $this->videoProvider->loadProviderFromInput($input);
+    return $video_provider->getName();
```
Comment (sherakama) [inline src/Plugin/BundleSuggestion/VideoEmbed.php]: That is nice of them to provide this interface. Handy.

**Comment by boznik** [comment]
@sherakama i was just about to take a test spin...

**Comment by pookmish** [comment]
test spin

**Comment by boznik** [comment]
The website encountered an unexpected error. Please try again later.
ArgumentCountError: Too few arguments to function Drupal\stanford_media\Plugin\BundleSuggestionManager::__construct(), 4 passed in /Users/ggarvey/www/suhumsci/docroot/core/lib/Drupal/Component/DependencyInjection/Container.php on line 274 and exactly 5 expected in Drupal\stanford_media\Plugin\BundleSuggestionManager->__construct() (line 51 of modules/custom/stanford_media/src/Plugin/BundleSuggestionManager.php).
Drupal\stanford_media\Plugin\BundleSuggestionManager->__construct(Object, Object, Object, Object) (Line: 274)
Drupal\Component\DependencyInjection\Container->createService(Array, 'plugin.manager.bundle_suggestion_manager') (Line: 171)
Drupal\Component\DependencyInjection\Container->get('plugin.manager.bundle_suggestion_manager') (Line: 67)
Drupal\stanford_media\Plugin\EntityBrowser\Widget\MediaBrowserBase::create(Object, Array, 'embed_code', Array) (Line: 21)
Drupal\Core\Plugin\Factory\ContainerFactory->createInstance('embed_code', Array) (Line: 83)
Drupal\Component\Plugin\PluginManagerBase->createInstance('embed_code', Array) (Line: 81)
Drupal\Core\Plugin\DefaultLazyPluginCollection->initializePlugin('ffe1badf-f391-4e5a-8fe4-da745c9da5af') (Line: 80)
Drupal\Component\Plugin\LazyPluginCollection->get('ffe1badf-f391-4e5a-8fe4-da745c9da5af') (Line: 17)
Drupal\entity_browser\WidgetsCollection->sortHelper('39f2b6ed-f27d-42d8-a161-004510188d8d', 'ffe1badf-f391-4e5a-8fe4-da745c9da5af')
uasort(Array, Array) (Line: 90)
Drupal\Core\Plugin\DefaultLazyPluginCollection->sort() (Line: 278)
Drupal\entity_browser\Entity\EntityBrowser->getWidgets() (Line: 192)
Drupal\entity_browser\Form\EntityBrowserForm->isFunctionalForm() (Line: 135)
Drupal\entity_browser\Form\EntityBrowserForm->buildForm(Array, Object)
call_user_func_array(Array, Array) (Line: 518)
Drupal\Core\Form\FormBuilder->retrieveForm('entity_browser_media_browser_form', Object) (Line: 275)
Drupal\Core\Form\FormBuilder->buildForm('entity_browser_media_browser_form', Object) (Line: 93)
Drupal\Core\Controller\FormController->getContentResult(Object, Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 582)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 151)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 68)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 99)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 78)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 52)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 693)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)

**Comment by pookmish** [comment]
Cache clear should fix that error

**Pull request merged**